### PR TITLE
WIP - add option to purchase hbar through Carbon

### PR DESCRIPTION
**Pull request by Sheng-Long**
Still need production key and a way to obscure it is supposed to be secret

### CODE REVIEW

File: src/components/ModalPurchaseHbar.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+} from "@vue/composition-api";
+import { store } from "../store";
+
+export default createComponent({
+    components: { Modal },
+    model: {
+        prop: "isOpen",
+        event: "change"
+    },
+    props: { isOpen: (Boolean as unknown) as PropType<boolean> },
```
Comment (cacampbell) [inline src/components/ModalPurchaseHbar.vue]: `Boolean`

File: src/components/ModalPurchaseHbar.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    setup() {
+        const accId = store.state.wallet.session!.account;
```
Comment (cacampbell) [inline src/components/ModalPurchaseHbar.vue]: `getters.GET_WALLET()`

or implement 

`getters.CURRENT_USER()`

or something like that

File: src/components/ModalPurchaseHbar.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        const accIdString = accId.toString();
+        const environment = process.env.NODE_ENV !== "production" ? "sandbox" : "production";
+        const key = process.env.NODE_ENV !== "production" ? "89fa28dd-b26e-4af4-8313-1536054767d5" : "production";
```
Comment (cacampbell) [inline src/components/ModalPurchaseHbar.vue]: huh?

The key (promo code) is `89...` for non-production and `production` for production?

Comment (Sheng-Long) [reply src/components/ModalPurchaseHbar.vue]: Just a placeholder for now

**Pull request merged**